### PR TITLE
Real-time Notifications

### PR DESCRIPTION
**Pull request by AdamDotCom**
### What does this do?

Part 1 introduces real-time notifications using Pusher to connect to our pre-existing Pusher events from the API to our Notifications. Part 2 will introduce browser events for these notifications.

### How do I test this?

- As User A open zOS, click on the Notifications to open the modal
- As User B open Zero Web
- As User B mention User A in a message
- As User A verify that a new notification has arrived in your Notifications modal

### CODE REVIEW

File: src/store/notifications/saga.test.ts [hunk 1]
```diff
@@ -119,4 +129,134 @@ describe('notifications list saga', () => {
...
+        expect(payload.pattern).toEqual('*');
+      })
+
+      .next({ userId: undefined })
+      .inspect((action) => {
+        expect(action).toEqual({ type: SagaActionTypes.CancelEventWatch });
+      })
+
+      .next();
+    // .isDone()
```
Comment (AdamDotCom) [inline src/store/notifications/saga.test.ts:229]: What happens with this test if I omit `.isDone`? This test is a non breaking while loop.

Comment (dalefukami) [reply src/store/notifications/saga.test.ts:229]: I've done it. I'm not entirely sure but even if the references to the unfinished generator hang around I can't imagine we'd over consume any memory or anything in our test suite. It'd take quite a bit, I think.

File: src/store/notifications/saga.ts [hunk 2]
```diff
@@ -21,13 +31,81 @@ export function* fetch(action) {
...
+    });
+
+    if (abort) {
+      notificationChannel.close();
+      return false;
+    }
+
+    if (relevantNotificationTypes.includes(notification.notificationType)) {
+      yield call(addNotification, notification);
+    }
```
Comment (dalefukami) [inline src/store/notifications/saga.ts:72]: I think we just add every notification? The denormalize already filters all the event types.

File: src/store/notifications/saga.ts [hunk 3]
```diff
@@ -21,13 +31,81 @@ export function* fetch(action) {
...
+}
+
 export function* clearNotifications() {
   yield all([
     put(removeAll({ schema: schema.key })),
     put(receive([])),
   ]);
 }
 
+export function* authWatcher() {
```
Comment (dalefukami) [inline src/store/notifications/saga.ts:94]: Something I forgot to test with my stuff....does this work even when you just switch wallet accounts via metamask? I.E., is there a logout/login combination?

Comment (AdamDotCom) [reply src/store/notifications/saga.ts:94]: When we switch addresses we `terminate` before `authorizing`. I did a smoke test with Pusher in debug mode and verified the client is disconnect / reconnecting with the new user.

```
Handling 'accountsChanged' event with payload ['0x4d054f8c925e992518861e0c1931b212aea2320d']
Emitting 'Web3ReactUpdate' with payload {account: '0x4d054f8c925e992518861e0c1931b212aea2320d'}
Pusher :  : ["State changed","connected -> disconnected"]
Pusher :  : ["State changed","initialized -> connecting"]
Pusher :  : ["Connecting",{"transport":"ws","url":"wss://ws-us2.pusher.com:443/app/8370d96dd06fa311e918?protocol=7&client=js&version=8.0.2&flash=false"}]
Pusher :  : ["State changed","connecting -> connected with new socket ID 23773.13929648"]
Pusher :  : ["Event sent",{"event":"pusher:subscribe","data":{"auth":"","channel":"notifications-ce60cf6c-baff-424c-9f5a-058d2ee023c4"}}]
Pusher :  : ["Event recd",{"event":"pusher_internal:subscription_succeeded","channel":"notifications-ce60cf6c-baff-424c-9f5a-058d2ee023c4","data":{}}]
Pusher :  : ["No callbacks on notifications-ce60cf6c-baff-424c-9f5a-058d2ee023c4 for pusher:subscription_succeeded"]
Pusher :  : ["Event sent",{"event":"pusher:subscribe","data":{"auth":"8370d96dd06fa311e918:de5c24adc35d4e5c4bf990d5d19718c1f28e42f6ac780117bfc3118806074575","channel_data":"{\"user_id\":\"ce60cf6c-baff-424c-9f5a-058d2ee023c4\"}","channel":"presence-channel"}}]
Pusher :  : ["Event recd",{"event":"pusher_internal:subscription_succeeded","channel":"presence-channel","data":{"presence":{"count":2,"ids":["7cb0000d-cf0c-4ca7-81ce-8ce72649c4fd","ce60cf6c-baff-424c-9f5a-058d2ee023c4"],"hash":{"7cb0000d-cf0c-4ca7-81ce-8ce72649c4fd":null,"ce60cf6c-baff-424c-9f5a-058d2ee023c4":null}}}}]
```

**Pull request merged**